services/kuber/server/domain/utils/nodes/patch.go

```diff
@@ -89,6 +89,27 @@ func (p *Patcher) PatchLabels() error {
 	return err
 }
 
+func (p *Patcher) PatchAnnotations() error {
+	var err error
+	for _, np := range p.desiredNodepools {
+		nodeAnnotations := np.Annotations
+		for _, node := range np.Nodes {
+			nodeName := strings.TrimPrefix(node.Name, fmt.Sprintf("%s-", p.clusterID))
+			for key, value := range nodeAnnotations {
+				patchPath, err1 := buildJSONPatchString("replace", "/metadata/annotations/"+key, value)
+				if err1 != nil {
```

Comment:
This will be repeated multiple times for each node. Doesn't this mean that only the last annotation will be applied ? since by calling 
`buildJSONPatchString("replace", "/metadata/annotations/"+key, value)` it will be replaced

Reply:
The replace address will be `/metadata/annotations/annotation_key` so only the annotation itself can be replaced. Its not replacing the whole `/metadata/annotations` each time. The same way labales are working atm.

Reply:
Yes, sorry I missed the `+key` part